src/main/java/com/rpg2014/invokers/EC2Invoker.java

```diff
@@ -1,24 +1,54 @@
 package com.rpg2014.invokers;
 
+import com.rpg2014.model.Ec2MethodNames;
 import com.rpg2014.wrappers.SpidermanEC2Wrapper;
+
+import javax.ws.rs.InternalServerErrorException;
 import java.lang.reflect.InvocationTargetException;
 import java.lang.reflect.Method;
+import java.util.Optional;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
 
 public class EC2Invoker {
 
-    private static SpidermanEC2Wrapper instance = SpidermanEC2Wrapper.getInstance();
+    Executor executor = Executors.newSingleThreadExecutor();
+    private SpidermanEC2Wrapper instance = SpidermanEC2Wrapper.getInstance();
+
+    public Optional invoke(Ec2MethodNames methodName) {
+        Runnable r;
+        switch (methodName) {
+            case StartInstance:
+            case StopInstance:
+                r = getAsyncRunnable(methodName);
+                executor.execute(r);
+                return Optional.empty();
+            default:
```

Comment:
since there's only four possible method names being passed it's better practice to just put all four and leave out the default case.

Reply:
true